src/odbc/sql2tds.c

```diff
@@ -237,11 +237,18 @@ odbc_convert_table(TDS_STMT *stmt, SQLTVP *src, TDS_TVP *dest, SQLLEN num_rows)
 	}
 
 	/* Convert the columns from a linked list to an array */
-	if ((cols = malloc(src->num_cols * sizeof(SQLTVPCOLUMN *))) == NULL)
+	if ((cols = calloc(src->num_cols, sizeof(SQLTVPCOLUMN *))) == NULL)
```

Comment:
You can also use the `tds_new0` utility. like `if ((cols = tds_new0(SQLTVPCOLUMN, src->num_cols)) == NULL)`